### PR TITLE
feat: Enable dual stack network

### PR DESCRIPTION
**Pull request by tregad**
# Adding options to set custom networks

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Applied my changes and created a cluster using a custom cluster and service network on rocky9.

### CODE REVIEW

**Comment by tregad** [comment]
Could be relevant for #122

**Comment by ahyattdev** [comment]
Tested and works for me with these options:

```yaml
    rke2_cluster_cidr:
      - 10.42.0.0/16
      - fd10::/56
    rke2_service_cidr:
      - 10.43.0.0/16
      - fd11::/108
```

A few caveats (from RKE2 itself):

- You must rebuild your cluster to change these IP ranges
- By default a /64 is allocated for each node. So ensure your cluster CIDR is large enough.
- Service CIDR must be no larger than a /108.

File: defaults/main.yml [hunk 1]
```diff
@@ -294,3 +294,9 @@ rke2_debug: false
 
 # The value for the node-name configuration item
 rke2_node_name: "{{ inventory_hostname }}"
+# default pod network range for rke2
```
Comment (MonolithProjects) [inline defaults/main.yml:297]: Please add empty line above the comments.

File: defaults/main.yml [hunk 2]
```diff
@@ -294,3 +294,9 @@ rke2_debug: false
...
+rke2_cluster_cidr:
+  - 10.42.0.0/16
+# default service network range for rke2
```
Comment (MonolithProjects) [inline defaults/main.yml:300]: Please add empty line above the comments.

**Review by MonolithProjects — COMMENTED** [review]
Please add new variables also to the README file

File: defaults/main.yml (hunk 2 above)
Comment (tregad) [reply defaults/main.yml:302]: Done.

**Review by MonolithProjects — APPROVED** [review]
LGTM. Thanks!

**Comment by adamcharnock** [comment]
FYI, this is a breaking change for configurations that previously set these values using `rke2_server_options`:

```yaml
    rke2_server_options:
      - "cluster-cidr: a.b.c.d/z"
      - "service-cidr: a.b.c.e/z"
```

In this case the cluster restarts with the default CIDRs configured, and not the ones above. I didn't see this mentioned in the release notes.

**Comment by MonolithProjects** [comment]
Hi @adamcharnock, it is mentioned in the release notes https://github.com/lablabs/ansible-role-rke2/releases/tag/1.36.0

**Comment by adamcharnock** [comment]
Ah indeed it is, my mistake. I was likely scanning for a 'breaking changes' heading, but that's on me, it is clearly there.

**Pull request merged**